### PR TITLE
Mx 15065 fixed order consensus

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
## Reasoning behind the pull request
- moving to 400 sized consensus group, it is no longer required to shuffle the order of signing, as every eligible node is required to sign.
- this simplifies the equivalent proof mechanics verification as it can be almost self contained during an epoch.
- this also makes it simpler to create consensus zk proofs
  
## Proposed changes
- make the order of signing fixed during the epoch, and use the ordering of nodes in the eligible list which is defined at the start of the epoch
- explicitly return the consensus leader/proposer
- add activation flag for the transition from dynamic ordering to fixed ordering

## Testing procedure
- regular system tests transitioning from dynamic ordering to fixed ordering, consensus & operation transitions successfully, blocks are added to the shard chains and notarization continues.
- rewards checks - rewards converge to the same value for each node over a longer period of time
- check number of selections as consensus leader for each node over a longer period of time, converges to the same value.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: common/constants.go [hunk 1]
```diff
@@ -1223,5 +1223,6 @@ const (
 	EGLDInESDTMultiTransferFlag                        core.EnableEpochFlag = "EGLDInESDTMultiTransferFlag"
 	CryptoOpcodesV2Flag                                core.EnableEpochFlag = "CryptoOpcodesV2Flag"
 	EquivalentMessagesFlag                             core.EnableEpochFlag = "EquivalentMessagesFlag"
+	FixedOrderInConsensusEnableEpochFlag               core.EnableEpochFlag = "FixedOrderInConsensusEnableEpoch"
```
Comment (sstanculeanu) [inline common/constants.go:1226]:
```suggestion
	FixedOrderInConsensusFlag               core.EnableEpochFlag = "FixedOrderInConsensusFlag"
```

File: go.mod [hunk 2]
```diff
@@ -4,6 +4,7 @@ go 1.20
 
 require (
 	github.com/beevik/ntp v1.3.0
+	github.com/btcsuite/btcutil v0.0.0-20190425235716-9e5f4b9a998d
```
Comment (sstanculeanu) [inline go.mod:7]: looks like it is not used anywhere.. `go mod tidy`

File: integrationTests/multiShard/block/interceptedHeadersSigVerification/interceptedHeadersSigVerification_test.go [hunk 3]
```diff
@@ -160,6 +162,7 @@ func TestInterceptedShardBlockHeaderWithLeaderSignatureAndRandSeedChecks(t *test
 		t.Skip("this is not a short test")
 	}
 
+	_ = logger.SetLogLevel("*:DEBUG")
```
Comment (sstanculeanu) [inline integrationTests/multiShard/block/interceptedHeadersSigVerification/interceptedHeadersSigVerification_test.go:165]: is this needed?

Comment (AdoAdoAdo) [reply integrationTests/multiShard/block/interceptedHeadersSigVerification/interceptedHeadersSigVerification_test.go:165]: forgot to remove

**Review by sstanculeanu — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

File: consensus/spos/consensusState.go [hunk 4]
```diff
@@ -151,7 +152,7 @@ func (cns *ConsensusState) GetLeader() (string, error) {
 		return "", ErrEmptyConsensusGroup
 	}
 
-	return cns.consensusGroup[IndexOfLeaderInConsensusGroup], nil
```
Comment (ssd04) [inline consensus/spos/consensusState.go:154]: remove `IndexOfLeaderInConsensusGroup`? as it should not be used anymore, it can be done on next PR

Comment (AdoAdoAdo) [reply consensus/spos/consensusState.go:154]: done in #6448

**Review by ssd04 — APPROVED** [review]

**Pull request merged**